src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt

```diff
@@ -244,34 +244,32 @@ class PrisonerIndexService(
     return totalRows
   }
 
-  fun withRestrictedPatientIfOut(booking: OffenderBooking): OffenderBooking {
-    if (booking.assignedLivingUnit?.agencyId != "OUT") return booking
-    val restrictivePatient = restrictedPatientService.getRestrictedPatient(booking.offenderNo) ?: return booking
-
-    return booking.copy(
-      locationDescription = booking.locationDescription + " - discharged to " + restrictivePatient.hospitalLocation.description,
-      restrictivePatient = RestrictivePatient(
-        supportingPrisonId = restrictivePatient.supportingPrison.agencyId,
-        dischargedHospital = restrictivePatient.hospitalLocation,
-        dischargeDate = restrictivePatient.dischargeTime.toLocalDate(),
-        dischargeDetails = restrictivePatient.commentText
-      )
-    )
-  }
+  fun OffenderBooking.getRestrictedPatientData(): RestrictivePatient? =
+    this.takeUnless { this.assignedLivingUnit?.agencyId != "OUT" }?.let {
+      restrictedPatientService.getRestrictedPatient(this.offenderNo)?.let {
+        RestrictivePatient(
+          supportingPrisonId = it.supportingPrison.agencyId,
+          dischargedHospital = it.hospitalLocation,
+          dischargeDate = it.dischargeTime.toLocalDate(),
+          dischargeDetails = it.commentText
+        )
+      }
+    }
```

Comment:
This is the offending function that "pretended" restrictive patients was in the prison-api domain object, so move this mapping to correct place